sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -50,31 +51,39 @@ static void skl_set_hda_codec_autosuspend_delay(struct snd_soc_card *card)
 
 #define IDISP_HDMI_BE_ID	1
 #define HDA_BE_ID		4
+#define SSP_AMP_BE_ID		5
 #define DMIC01_BE_ID		6
 #define DMIC16K_BE_ID		7
 #define BT_OFFLOAD_BE_ID	8
 
 #define HDA_LINK_ORDER	SOF_LINK_ORDER(SOF_LINK_IDISP_HDMI,  \
 				       SOF_LINK_HDA,        \
+				       SOF_LINK_AMP,        \
 				       SOF_LINK_DMIC01,     \
 				       SOF_LINK_DMIC16K,    \
 				       SOF_LINK_BT_OFFLOAD, \
-				       SOF_LINK_NONE,       \
 				       SOF_LINK_NONE)
 
 #define HDA_LINK_IDS	SOF_LINK_ORDER(IDISP_HDMI_BE_ID,  \
 				       HDA_BE_ID,        \
+				       SSP_AMP_BE_ID,    \
 				       DMIC01_BE_ID,     \
 				       DMIC16K_BE_ID,    \
 				       BT_OFFLOAD_BE_ID, \
-				       0,                \
 				       0)
 
 static unsigned long
 skl_hda_get_board_quirk(struct snd_soc_acpi_mach_params *mach_params)
 {
 	unsigned long board_quirk = 0;
 	int ssp_bt;
+	int ssp_amp;
+
+	if (mach_params->i2s_link_mask) {
+		ssp_amp = fls(mach_params->i2s_link_mask) - 1;
+		if (ssp_amp >= 0)
+			board_quirk |= SOF_SSP_PORT_AMP(ssp_amp);
```

Comment:
Not sure if it is always true. How do you know that is an AMP not something else?